Latest Clang finds -Wtautological-compare warning

```
external/libevent/http.c:3486:18: error: pointer comparison always evaluates to false [-Werror,-Wtautological-compare]
 3134 |                 if (uri + slen < uri) {
      |
```

--- a/http.c
+++ b/http.c
@@ -3483,7 +3483,7 @@ evhttp_uriencode(const char *uri, ev_ssize_t len, int space_as_plus)
 			goto out;
 		}
 
-		if (uri + slen < uri) {
+		if (slen > (size_t)(UINTPTR_MAX - (uintptr_t)uri)) {
 			goto out;
 		}
 

